### PR TITLE
Provide and initial ADO bootstrap template to help users get stared with IaC pipelines

### PR DESCRIPTION
**Pull request by ThePhen**
## All Submissions:
-------------------------------------
* [YES] Have you followed the guidelines in our Contributing [document](./CONTRIBUTING.md)?
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [NA] I have added tests to cover my changes.
* [NA] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## What is the current behavior?
-------------------------------------
Issue Number: #373 

## What is the new behavior?
-------------------------------------
Provides a basic 'bootstrap' template for Cobalt users

## Does this introduce a breaking change?
-------------------------------------
- [NO]

### CODE REVIEW

File: infra/templates/ado-bootstrap-iac-pipeline/.gitignore [hunk 1]
```diff
@@ -0,0 +1,32 @@
+.envrc
```
Comment (nmiodice) [inline infra/templates/ado-bootstrap-iac-pipeline/.gitignore:1]: Should we have this in the top level `.gitignore` file?

File: infra/templates/ado-bootstrap-iac-pipeline/README.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
+# Bootstrap for Cobalt IaC Pipelines on Azure DevOps
+
+## Overview 
+This provides a starting point for anyone interested in having an Azure DevOps
+Project that will 'run' a target Cobalt template whenever that template changes.
+
+## Intended audience
+
+*FIXME:* Application developer that is brand new to Cobalt templating and it's *Cobalt Infrastructure Template* (CIT) developer workflow.
```
Comment (nmiodice) [inline infra/templates/ado-bootstrap-iac-pipeline/README.md:9]: Did you intend to keep the *FIXME* remark (and the others?)

File: infra/templates/ado-bootstrap-iac-pipeline/README.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+### An example
```
Comment (nmiodice) [inline infra/templates/ado-bootstrap-iac-pipeline/README.md:11]: I *LOVE* this section

File: infra/templates/ado-bootstrap-iac-pipeline/README.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+## What the Bootstrap Template needs, provisions, and outputs
+
+To work properly, the Bootstrap Template needs some information about how you want things named, how many environments
+are desired (and their details), and little else. The template also creates many kinds of resources
+
+### Decide with Project to use
+
+Visit your ADO Organization site and select or create a project. This project will house the pipelines that
+will operate your IaC CI/CD pipelines. Note that project's name. We'll need to use it as an input, in the next section.
```
Comment (nmiodice) [inline infra/templates/ado-bootstrap-iac-pipeline/README.md:52]: May be worthwhile to point out that if the user wants to create the project as well, then they can modify the template to do so. perhaps the implementation of that might live in a comment alongside the project `data` source?

File: infra/templates/ado-bootstrap-iac-pipeline/README.md (hunk 2 above)
Comment (nmiodice) [reply infra/templates/ado-bootstrap-iac-pipeline/README.md:9]: I forgot to add this in my review, but I think that the only content I see that is missing from this document is the following:

- A note/docs ref about setting up the ADO provider locally
- What to do after this template (fork cobalt, push to repo that was created, and then trigger a build)

Have you tested the latter to make sure it works?

File: infra/templates/ado-bootstrap-iac-pipeline/.gitignore (hunk 1 above)
Comment (ThePhen) [reply infra/templates/ado-bootstrap-iac-pipeline/.gitignore:1]: Yes, but I haven't run across it's usage elsewhere. More generally, should all of this files contents be in the root .gitignore for the repo?

**Comment by ThePhen** [comment]
Thanks for the look, @nmiodice. I really did want to pull in the 'create AzDO Project' parts of the template... I'll add them,

File: infra/templates/ado-bootstrap-iac-pipeline/.gitignore (hunk 1 above)
Comment (nmiodice) [reply infra/templates/ado-bootstrap-iac-pipeline/.gitignore:1]: Probably, though this is based purely on what I've done in the past, and less about any best practice. your call.

File: infra/templates/ado-bootstrap-iac-pipeline/README.md (hunk 3 above)
Comment (ThePhen) [reply infra/templates/ado-bootstrap-iac-pipeline/README.md:11]: +1

File: infra/templates/ado-bootstrap-iac-pipeline/.gitignore (hunk 1 above)
Comment (ThePhen) [reply infra/templates/ado-bootstrap-iac-pipeline/.gitignore]: I migrate the ignores up to the root.

File: infra/templates/ado-bootstrap-iac-pipeline/README.md (hunk 2 above)
Comment (ThePhen) [reply infra/templates/ado-bootstrap-iac-pipeline/README.md]: Tested!
image

**Pull request merged**